compiler/cpp/src/generate/t_java_generator.cc

```diff
@@ -3967,52 +3980,58 @@ string t_java_generator::function_signature_async(t_function* tfunction, bool us
  * @param tfunction Function definition
  * @return String of rendered function definition
  */
-string t_java_generator::function_signature_future(t_function* tfunction, string prefix) {
+string t_java_generator::function_signature_future(t_function* tfunction, int params_futures_mask, string prefix) {
   t_type* ttype = tfunction->get_returntype();
   std::string return_type = type_name(ttype, true);
   std::string fn_name = get_rpc_method_name(tfunction->get_name());
-  std::string arg_list = argument_list(tfunction->get_arglist());
+  std::string arg_list = argument_list(tfunction->get_arglist(), true, params_futures_mask);
   return "CompletableFuture<" + return_type + "> " + prefix + fn_name + "(" + arg_list + ")";
 }
 
-string t_java_generator::async_function_call_arglist(t_function* tfunc, bool use_base_method, bool include_types) {
```

Comment:
Argh. Accident, I modified the future argument list which was in a different place, instead of the normal argument list. I'll move it back so it disappears form the diff.